README.md

```diff
@@ -1,5 +1,181 @@
+2. Clone this repo on your local computer. In the newly-created directory (robosub-ros), run the landside container. If on Windows, use PowerShell.
+    ```bash
+    docker run -td -p 2201:2201 -v ${PWD}:/root/dev/robosub-ros dukerobotics/robosub-ros:landside
+    ```
+
+3. Connect to the landside container using SSH.
+    ```bash
+    ssh -XY -p 2201 root@localhost
+    ```
+4. Connect to the onboard container using SSH.
+    ```bash
+    ssh -p 2200 root@192.168.1.1
+    ```
+
+### Local Testing
+Use these instructions to test code on your computer by simulating the robot's execution.
+
+1. To run the containers, clone this repo. In the newly-created directory (`robosub-ros`), execute
+    ```bash
+    docker-compose up -d
+    ```
+    This will pull the images if you don't have them, create a new network that simulates the network we use on our robot, mount the code, and start the containers.
+    
+    To update the images, or to just pull them without running them, use `docker-compose pull`.
+
+2. To connect into the containers, you can SSH into them. 
+
+    For the onboard container, use the command 
+    ```bash
+    ssh -p 2200 root@localhost
+    ```
+
+    For the landside container, use the command
+    ```bash
+    ssh -XY -p 2201 root@localhost
+    ```
+
+3. To stop and delete both containers and their network, in the `robosub-ros` directory, execute
+    ```bash
+    docker-compose down
+    ```
+
+
+## Running Our Code 
+
+We use a custom build script that simplifies the process of building and overlaying catkin workspaces.
+
+:warning: All commands should be run inside of a Docker container.
+ 
+### Executing Build Script 
+To build our workspaces for ROS, in the `robosub-ros` directory, execute
+```bash
+./build.sh <workspace>
+```
+where `<workspace>` is the workspace to build, either `onboard` or `landside`.
+
+### Sourcing Setup File
+Once the build script has finished executing, source the setup file using
+```bash
+source <workspace>/catkin_ws/devel/setup.bash
+```
+where `<workspace>` is the workspace you just built. This will also be the last line that is printed from the build script, so you can also just copy and execute that.
+
+You should now be ready to use our packages and ROS.
+
+### Running Launch Files
+Here are some common launch configurations for both pool and local testing.
+
+* Run controls (simulation)
+    ```bash
+    roslaunch execute motion.launch sim:=true
+    rosrun controls test_state_publisher.py
+    rosservice call \enable_controls true
+    ```
+* Run controls (pool test)
+    ```bash
+    roslaunch execute motion.launch
+    rosrun controls test_state_publisher.py
+    rosservice call \enable_controls true
+    ```
+* Pool Test with Joystick (see [Joystick Documentation](https://github.com/DukeRobotics/robosub-ros/tree/master/landside/catkin_ws/src/joystick))
+    ```bash
+    ```
+* Useful commands for testing
+    * Reset State (to zeroes)
+        ```bash
+        rosservice call 
```

Comment:
Didn't remember the full service call for resetting the state, something to ask David about.

Reply:
rosservice call /set_pose “pose:
 header:
  seq: 0
  stamp:
   secs: 0
   nsecs: 0
  frame_id: ‘’
 pose:
  pose:
   position: {x: 0.0, y: 0.0, z: 0.0}
   orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}
  covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
   0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
   0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]”

If you tab complete /set_pose you need to manually change w from 0.0 to 1.0

Reply:
I added both options in as compact as I could. Let me know if it's not clear and/or if we should cut it down somehow (the full command takes up a lot of space). I'm ok with anything.
(also curious what @muthuArivoli thinks).

Reply:
I guess it's fine. You should get someone new to create a service that just resets the state.